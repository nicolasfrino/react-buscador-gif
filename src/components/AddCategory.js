import React, {useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputvalue, setinputvalue] = useState('');

    const handleInputChange = (e) => {
        setinputvalue(e.target.value);

        //console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log('handleSubmit'); 

        if (inputvalue.trim().length > 2) {
            setCategories(cats => [ inputvalue, ...cats]); //cats => ['Pokemon', ...categories]
            setinputvalue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="Text"
                value={inputvalue}
                //onChange={ (e) => console.log(e)}
                onChange={ handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;