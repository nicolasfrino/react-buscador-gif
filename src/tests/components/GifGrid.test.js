import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const categoria = 'Goku';

    test ('debe mostrarse OK', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper =  shallow ( <GifGrid  category={categoria} /> );
        expect( wrapper ).toMatchSnapshot();

    })

    test ('debe mostrar items cuando se llama a userFetchGifts', () => {


        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiera.jpg',
            title: 'cualquiera'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });


        const wrapper =  shallow ( <GifGrid  category={categoria} /> );

        //como hacer para no comparar contra snapshot y mirar si esta el componente
        expect( wrapper ).toMatchSnapshot();

        expect (wrapper.find('p').exists()).toBe(false);

        expect (wrapper.find('GifGridItem').length).toBe( gifs.length);

    })


})
