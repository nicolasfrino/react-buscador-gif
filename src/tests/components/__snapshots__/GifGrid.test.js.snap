// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se llama a userFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquiera"
      url="https://localhost/cualquiera.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse OK 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
